types/lodash/lodash-tests.ts

```diff
@@ -11167,60 +11167,77 @@ namespace TestFunctionsIn {
 // _.get
 namespace TestGet {
     {
-        let result: string;
+        // when object is empty - defenitely default value will be returned
+        _.get(null, 'a'); // $ExpectType undefined
+        _.get(null, 'a', 1); // $ExpectType 1
+    }
+    {
+        // when depth of object is 1 and path to value is right - or value, or default will be returned (in case value is undefined)
+        const objWithFoo: { foo: number } = any;
 
-        result = _.get('abc', '0');
-        result = _.get('abc', '0', '_');
-        result = _.get('abc', ['0']);
-        result = _.get('abc', ['0'], '_');
+        _.get(objWithFoo, 'foo'); // $ExpectType number
+        _.get(objWithFoo, ['foo']); // $ExpectType number
+        _.get(objWithFoo, 'foo', false); // $ExpectType number | false
+        _.get(objWithFoo, ['foo'], false); // $ExpectType number | false
+        _.get(objWithFoo, 'foo', 1); // $ExpectType number
+        _.get(objWithFoo, ['foo'], 1); // $ExpectType number
 
-        result = _('abc').get('0');
-        result = _('abc').get('0', '_');
-        result = _('abc').get(['0']);
-        result = _('abc').get(['0'], '_');
-    }
+        _(objWithFoo).get('foo'); // $ExpectType number
+        _(objWithFoo).get(['foo']); // $ExpectType number
+        _(objWithFoo).get('foo', false); // $ExpectType number | false
+        _(objWithFoo).get(['foo'], false); // $ExpectType number | false
+        _(objWithFoo).get('foo', 1); // $ExpectType number
+        _(objWithFoo).get(['foo'], 1); // $ExpectType number
 
-    {
-        let result: number;
+        const objWithMaybeFoo: { foo?: number } = any;
 
-        result = _.get([42], '0');
-        result = _.get([42], '0', -1);
-        result = _.get([42], ['0']);
-        result = _.get([42], ['0'], -1);
+        _.get(objWithMaybeFoo, 'foo'); // $ExpectType number | undefined
+        _.get(objWithMaybeFoo, ['foo']); // $ExpectType number | undefined
+        _.get(objWithMaybeFoo, 'foo', false); // $ExpectType number | false | undefined
```

Comment:
See here, the return type should really be `number | false`. But I can't think of a good way to do that, so maybe `number | false | undefined` is good enough.